Failed: akka.stream.scaladsl.RestartSpec

https://jenkins.akka.io:8498/job/akka-nightly/4657/testReport/junit/akka.stream.scaladsl/RestartSpec/A_restart_with_backoff_flow_should_restart_on_cancellation/

```
java.lang.AssertionError: Expected OnNext(_), yet no element signaled during 3 seconds
      at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:290)
      at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:281)
      at akka.stream.scaladsl.RestartSpec$$anonfun$3$$anonfun$apply$mcV$sp$14.apply(RestartSpec.scala:362)
      at akka.stream.scaladsl.RestartSpec$$anonfun$3$$anonfun$apply$mcV$sp$14.apply(RestartSpec.scala:351)
```

https://jenkins.akka.io:8498/job/akka-nightly/4660/testReport/junit/akka.stream.scaladsl/RestartSpec/A_restart_with_backoff_flow_should_restart_on_completion/

```
 java.lang.AssertionError: Expected OnNext(_), yet no element signaled during 3 seconds
      at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:290)
      at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:281)
      at akka.stream.scaladsl.RestartSpec$$anonfun$3$$anonfun$apply$mcV$sp$15.apply(RestartSpec.scala:386)
      at akka.stream.scaladsl.RestartSpec$$anonfun$3$$anonfun$apply$mcV$sp$15.apply(RestartSpec.scala:373)
```

--- a/akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala
+++ b/akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala
@@ -298,7 +298,9 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
       override def onPull() = if (isAvailable(in)) {
         sourceOut.push(grab(in))
       } else {
-        pull(in)
+        if (!hasBeenPulled(in)) {
+          pull(in)
+        }
       }
       override def onDownstreamFinish() = {
         if (finishing) {

